test/Tests/data/xml/nested.xml

```diff
@@ -0,0 +1,11 @@
+<?xml version = "1.0"?>
+<foo id="1">
+    <bar id="2">
+    </bar>
+    <baz id="3">
+        <bar id="4">
+        </bar>
+    </baz>
+    <bar id="5">
+    </bar>
```

Comment:
I think I'd like to also see a `bar` inside of `bar` - I'm wondering if then both inner and outer are picked or is it only outer?

Reply:
Added this to test. It gets both inner and outer.